internal/api/controllers/private/runsCancelActions.go

```diff
@@ -0,0 +1,57 @@
+package private
+
+import (
+	"net/http"
+	"playbook-dispatcher/internal/api/controllers/public"
+	"playbook-dispatcher/internal/api/dispatch"
+	"playbook-dispatcher/internal/common/model/generic"
+	"playbook-dispatcher/internal/common/utils"
+
+	"github.com/google/uuid"
+	"github.com/spf13/viper"
+)
+
+func CancelInputV2GenericMap(
+	cancelInput CancelInputV2,
+	account string,
+	runId *uuid.UUID,
+	cfg *viper.Viper,
+) generic.CancelInput {
+	orgIdString := string(cancelInput.OrgId)
+	principal := string(cancelInput.Principal)
+
+	result := generic.CancelInput{
+		Account:   account,
+		RunId:     runId,
+		OrgId:     &orgIdString,
+		Principal: &principal,
+	}
+
+	return result
+}
+
+func runCancelError(code int) *RunCanceled {
+	return &RunCanceled{
+		Code: code,
+		// TODO report error
+	}
+}
+
+func handleRunCancelError(err error) *RunCanceled {
+	if _, ok := err.(*dispatch.RecipientNotFoundError); ok {
+		return runCancelError(http.StatusNotFound)
```

Comment:
Unlike the dispatch operation, where we can blame the caller for trying to run a playbook on an unknown host, with cancel we know the host was connected at some point (as long as the run entity exists). Therefore, I think we should use a different code other than 404 (as we know the connection existed just is not available at this point). Not sure which code, though.

Reply:
@jharting I went ahead and changed this to a 409 error code since it seemed to encompass the use case of a changed run state the best, let me know what you think.

Reply:
Sounds good to me